plugins/woocommerce/client/admin/client/settings-payments/onboarding/providers/woopayments/index.tsx

```diff
@@ -36,7 +37,7 @@ export default function WooPaymentsModal( {
 	// Open modal when on an onboarding route
 	React.useEffect( () => {
 		if (
-			location.pathname.startsWith( wooPaymentsOnboardingPath ) &&
+			location.pathname.endsWith( wooPaymentsOnboardingPath ) &&
```

Comment:
I'll look into how to make this more robust. We should probably compare the arg explicitly instead of the whole pathname.